src/us/kbase/narrativejobservice/sdkjobs/SDKMethodRunner.java

```diff
@@ -761,10 +761,16 @@ public static JobState checkJob(String jobId, AuthToken authPart,
 				Tuple3<String, String, String>, Tuple3<Long, Long, String>,
 				Long, Long, Tuple2<String, String>, Map<String, String>,
 				String, Results> j : ujsClient.listJobs2(new ListJobsParams().withAuthstrat("kbaseworkspace").withAuthparams(authParams))) {
-
-				String jobId = j.getE1();
-				JobState njsJobState = checkJobCondor(jobId, authPart, config);
-				js.add(njsJobState);
+					Tuple7<String, String, String, Long, String, Long, Long> jobStatus = new Tuple7<String, String, String, Long, String, Long, Long>();
+					jobStatus.setE1(j.getE6().getE2());
+					jobStatus.setE2(j.getE4());
+					jobStatus.setE3(j.getE5());
+					jobStatus.setE4(j.getE7().getE1());
+					jobStatus.setE5(j.getE6().getE3());
+					jobStatus.setE6(j.getE8());
+					jobStatus.setE7(j.getE9());
+					JobState njsJobState = checkJobCondor(j.getE1(), jobStatus, authPart, config);
```

Comment:
So this uses the 2nd overloaded checkjob function (checkjob_WITHSTATUS), and bypasses looking up UJS again, since UJS was looked up once.. And since this job status is created here, it doesn't need to be created in the new (checkjob_WITHSTATUS).. which means you don't need to edit the function signature for the API, and you don't need to duplicate the logic of checkjob in this listStatusByWorkspace function